### PR TITLE
Refactors for Fix/304 manual restart fails

### PR DESCRIPTION
**Pull request by kellpossible**
Refactors in preparation for solution to #304 

+ Remove storage lock and coordinator state lock. Rely instead on Rust's mutability analysis and management, the library user is responsible for managing this.
+ Remove unnecessary box on signature.
+ Move the responsibility for shutdown signal handling to the library user and update example with tokio signal handling for shutdown.

This isn't the final solution, it was a big refactor in the end so I put it into its own pull request. Haven't run all the tests yet, and there is still a compile error in there, but the example coordinator `main.rs` runs.

### CODE REVIEW

**Review by julesdesmit — COMMENTED** [review]
I really like the idea of using the compiler's mutability analysis instead of the mutexes we had before.

For now, I don't have any points to raise about specific lines, however, I'd like to see the current errors fixed so the tests can be ran. I'd like to give this a spin on the integration test too - unless it is missing some stuff, due to it being a partial fix for the mentioned issue? It looks fine to me, however.

**Review by apruden2008 — APPROVED** [review]

**Review by julesdesmit — APPROVED** [review]

**Comment by kellpossible** [comment]
This will be ready to merge when https://github.com/AleoHQ/aleo-setup-coordinator/pull/59 gets approved

**Review by emmorais — APPROVED** [review]
Good to see this PR. Simplification will help a lot.

File: phase1-coordinator/src/main.rs [hunk 1]
```diff
@@ -26,34 +29,51 @@ pub async fn main() -> anyhow::Result<()> {
...
 
-    ceremony.await.expect("The ceremony handle has panicked");
+    tokio::select! {
+        _ = shutdown_handler => {
+            println!("Shutdown completed first")
+        }
+        _ = ceremony => {
+            println!("Ceremony completed first")
+        }
+    };
```
Comment (ibaryshnikov) [inline phase1-coordinator/src/main.rs:76]: In docs for `select!`
> Wait on multiple concurrent branches, returning when the first branch completes, cancelling the remaining branches.

I don't know how often do we plan to kill the process, but it's a potential source of bugs. The futures in select should be cancellation-safe. So far we did nothing to ensure coordinator has this quality.

**Pull request merged**